models/tests/test_volunteer.py

```diff
@@ -58,6 +61,30 @@ def test_invalid_phone_too_short(self):
             Volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
                       phone='+12345')
 
+    def tearDown(self):
+        """
+        Function to delete stored data after a test has finished running
+        to avoid corrupting other tests
+        """
+        Plan.delete_all()
+
+    def test_adding_volunteer_to_camp(self):
+        """
+        Test to check functionality for adding a volunteer to a camp.
+        """
+        Plan(name='test_plan1',
+             emergency_type=Plan.EmergencyType.EARTHQUAKE,
+             description='Test emergency plan',
+             geographical_area='London',
+             camps=[Camp(name='camp1')])
+        volunteer_a = Volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                phone='+447519953189')
+        test_plan = Plan.find('test_plan1')
+        test_camp = test_plan.camps.get('camp1')
+        test_camp.volunteers.add(volunteer_a)
```

Comment:
Thank you! Unit test has been simplified per your suggestion and has passed in pycharm.

I also checked whether it was possible to create a camp not added to a plan and I managed it. Do you have any recommendations for what I should do to stop this? From what I remember, it's possible to do this because you need to create a camp in order to add it to a plan (so that a plan can satisfy the condition that it must be created with at least one camp, and so a camp must exist beforehand) so I'm not sure what I can do to handle the case where camp is created and then not assigned. Any advice?

Reply:
I suspect that creating a detached camp won't cause any error. 
Of course it doesn't make sense for the real code, but it should be fine for this test.

Reply:
Just found out that that will in fact cause an error, please ignore my comment